Update for tyk-mdcb-data-plane GA (#3052)
TykTechnologies/tyk-docs: tyk-docs/content/tyk-cloud/environments-deployments/hybrid-gateways-helm.md

description: "How to deploy Hybrid Data Plane on Kubernetes using new Helm Chart"
menu:
  main:
    parent: "Deploy Hybrid Gateways"
weight: 5
---

## New Tyk Helm Charts

Tyk is working to provide a new set of Helm charts, and will progressively roll them out at [tyk-charts](https://github.com/TykTechnologies/tyk-charts). It will provide component charts for all Tyk Components, as well as umbrella charts as reference configurations for open source and Tyk Self Managed users.

### Status of the New Charts

| Umbrella Charts | Description | Status |
|-----------------|-------------|--------|
| tyk-oss                | Tyk Open Source | Stable              |
| tyk-single-dc          | Tyk Self Managed (Single DC) | Beta            |
| tyk-mdcb-control-plane | Tyk Self Managed (MDCB) Control Plane | Coming Soon     |
| tyk-mdcb-data-plane    | Tyk Self Managed (MDCB) Data Plane <br> Tyk Hybrid Data Plane | Stable              |

To deploy hybrid data planes using the new Helm chart, please use [tyk-mdcb-data-plane](https://github.com/TykTechnologies/tyk-charts/tree/main/tyk-mdcb-data-plane) chart.

## Tyk MDCB Data Plane

> Tyk-mdcb-data-plane provides the default deployment of a Tyk data plane for Tyk Self Managed MDCB or Tyk Cloud users. It will deploy the data plane components that remotely connect to a MDCB control plane.

Reviewer: `Tyk-mdcb-data-plane` this part looks a bit unformatted in the preview. Should we quote it?

Author: ```suggestion
`tyk-mdcb-data-plane` provides the default deployment of a Tyk data plane for Tyk Self Managed MDCB or Tyk Cloud users. It will deploy the data plane components that remotely connect to a MDCB control plane.
```

2. Copy Key, Org ID, and Data Planes Connection String (MDCB) as `global.remoteControlPlane`'s `userApiKey`, `orgId`, and `connectionString` respectively.

{{< img src="/img/tyk-charts/tyk-cloud-deployment.png" alt="tyk-cloud-deployment" >}}

## Installing the Chart

To install the chart from the Helm repository in namespace `tyk` with the release name `tyk-data-plane`:

```bash
    helm repo add tyk-helm https://helm.tyk.io/public/helm/charts/
    helm repo update
    helm show values tyk-helm/tyk-mdcb-data-plane > values-data-plane.yaml
```

See [Configuration](#configuration) section for the available config options and modify your local `values-data-plane.yaml` file accordingly. Then install the chart:

```bash
    helm install tyk-data-plane tyk-helm/tyk-mdcb-data-plane -n tyk --create-namespace -f values-data-plane.yaml
```

## Uninstalling the Chart

```bash
helm uninstall tyk-data-plane -n tyk
```

This removes all the Kubernetes components associated with the chart and deletes the release.

## Upgrading Chart

```bash
helm upgrade tyk-data-plane tyk-helm/tyk-mdcb-data-plane -n tyk
```

> *Note: tyk-hybrid chart users*

Author: updated

Copy the following commands to add it:

```bash
helm upgrade tyk-redis oci://registry-1.docker.io/bitnamicharts/redis -n tyk --create-namespace --install --set image.tag=6.2.13
```

Follow the notes from the installation output to get connection details and password.

```
  Redis(TM) can be accessed on the following DNS names from within your cluster:

    tyk-redis-master.tyk.svc.cluster.local for read/write operations (port 6379)
    tyk-redis-replicas.tyk.svc.cluster.local for read-only operations (port 6379)

  export REDIS_PASSWORD=$(kubectl get secret --namespace tyk tyk-redis -o jsonpath="{.data.redis-password}" | base64 --decode)
```

The Redis address as set by Bitnami is `tyk-redis-master.tyk.svc.cluster.local:6379`

You can reference the password secret generated by Bitnami chart by  `--set global.redis.passSecret.name=tyk-redis` and `--set global.redis.passSecret.keyName=redis-password`, or just set `global.redis.pass=$REDIS_PASSWORD`

#### Evaluation only: via *simple-redis* chart

{{< warning  success >}}
**Warning**

Another option for Redis, to get started quickly, is to use our [simple-redis](https://artifacthub.io/packages/helm/tyk-helm/simple-redis) chart.

> Please note that these provided charts must never be used in production or for anything

Author: updated

3. Set `tyk-gateway.gateway.tls.useDefaultTykCertificate` to false.
4. Set `tyk-gateway.gateway.tls.secretName` to the name of the newly created secret.

*Add Custom Certificates*

To add your custom Certificate Authority(CA) to your containers, you can mount your CA certificate directly into /etc/ssl/certs folder.

```yaml
   extraVolumes: 
     - name: self-signed-ca
       secret:
         secretName: self-signed-ca-secret
   extraVolumeMounts: 
     - name: self-signed-ca
       mountPath: "/etc/ssl/certs/myCA.pem"
       subPath: myCA.pem
```

#### Accessing Gateway

*Service port*

Default service port of gateway is 8080. You can change this at `global.servicePorts.gateway`.

*Ingress*

An Ingress resource is created if `tyk-gateway.gateway.ingress.enabled` is set to true.

```yaml
    ingress:
      # if enabled, creates an ingress resource for the gateway
      enabled: true

      # specify ingress controller class name
      className: "nginx"

      # annotations for ingress
      annotations: {}

      # ingress rules
      hosts:
        - host: tyk-gw.local
          paths:
            - path: /
              pathType: ImplementationSpecific

>       # tls configuration for ingress

Author: the other section is removed for clarity

>       # tls configuration for ingress

Author: updated

* Kubernetes 1.19+
* Helm 3+
* Redis should already be installed or accessible by the gateway. For Redis installations instruction, follow the [Redis installation](#set-redis-connection-details-required) guide below.
* Connection details to remote control plane. See [Quick Start](#obtain-your-remote-control-plane-connection-details) on how to obtain them from Tyk Cloud.

## Quick Start

Quick start using `tyk-mdcb-data-plane` and Bitnami Redis chart

```bash
NAMESPACE=tyk
APISecret=foo
MDCB_UserKey=9d20907430e440655f15b851e4112345
MDCB_OrgId=64cadf60173be90001712345
MDCB_ConnString=mere-xxxxxxx-hyb.aws-euw2.cloud-ara.tyk.io:443
MDCB_GroupId=dc-uk-south

helm upgrade tyk-redis oci://registry-1.docker.io/bitnamicharts/redis -n $NAMESPACE --create-namespace --install --set image.tag=6.2.13

helm upgrade hybrid-dp tyk-helm/tyk-mdcb-data-plane -n $NAMESPACE --create-namespace \
  --install \
  --set global.remoteControlPlane.enabled=true \
  --set global.remoteControlPlane.userApiKey=$MDCB_UserKey \
  --set global.remoteControlPlane.orgId=$MDCB_OrgId \
  --set global.remoteControlPlane.connectionString=$MDCB_ConnString \
  --set global.remoteControlPlane.groupID=$MDCB_GroupId \

>   --set global.remoteControlPlane.useSSL=true \

Reviewer: As mentioned in https://github.com/TykTechnologies/tyk-docs/pull/3052/files#r1284155987, `useSSL` is enabled by default. Maybe we can simplify this helm command by removing default values in the command.

>   --set global.remoteControlPlane.sslInsecureSkipVerify=true \

Reviewer: As mentioned in https://github.com/TykTechnologies/tyk-docs/pull/3052/files#r1284155987, `useSSL` and `sslInsecureSkipVerify` are enabled by default. Maybe we can simplify this helm command by removing default values in the command.

Author: ```suggestion
```

* Kuberentes 1.19+
* Helm 3+
* Redis should already be installed or accessible by the gateway. For Redis installations instruction, follow the [Redis installation](#set-redis-connection-details-required) guide below.
* Connection details to remote control plane. See [Quick Start](#obtain-your-remote-control-plane-connection-details) on how to obtain them from Tyk Cloud.

## Quick Start

Quick start using `tyk-mdcb-data-plane` and Bitnami Redis chart

```bash
NAMESPACE=tyk
APISecret=foo
MDCB_UserKey=9d20907430e440655f15b851e4112345
MDCB_OrgId=64cadf60173be90001712345
MDCB_ConnString=mere-xxxxxxx-hyb.aws-euw2.cloud-ara.tyk.io:443
MDCB_GroupId=dc-uk-south

helm upgrade tyk-redis oci://registry-1.docker.io/bitnamicharts/redis -n $NAMESPACE --create-namespace --install --set image.tag=6.2.13

helm upgrade hybrid-dp tyk-helm/tyk-mdcb-data-plane -n $NAMESPACE --create-namespace \
  --install \

>   --set global.remoteControlPlane.enabled=true \

Reviewer: in mdcb-data-plane chart, `global.remoteControlPlane.enabled` is enabled by default. maybe, let's remove it - so, the first instruction to install the chart looks cleaner. What do you think?>

Author: ```suggestion
```

  --set global.remoteControlPlane.userApiKey=$MDCB_UserKey \
  --set global.remoteControlPlane.orgId=$MDCB_OrgId \
  --set global.remoteControlPlane.connectionString=$MDCB_ConnString \
  --set global.remoteControlPlane.groupID=$MDCB_GroupId \
  --set global.remoteControlPlane.useSSL=true \
  --set global.remoteControlPlane.sslInsecureSkipVerify=true \
  --set global.secrets.APISecret="$APISecret" \
  --set global.redis.addrs="{tyk-redis-master.$NAMESPACE.svc.cluster.local:6379}" \
  --set global.redis.passSecret.name=tyk-redis \
  --set global.redis.passSecret.keyName=redis-password
```

Gateway is now accessible through service `gateway-svc-hybrid-dp-tyk-gateway` at port `8080`.

Pump is also configured with Hybrid Pump which sends aggregated analytics to MDCB or Tyk Cloud, and Prometheus Pump which expose metrics locally at `:9090/metrics`.

### Obtain your Remote Control Plane Connection Details

You can easily obtain your remote control plane connection details on Tyk Cloud.

1. Go to Deployment tab and create a Hybrid data plane configuration. You can also select from an existing one.

> 2. Copy Key, Org ID, and Data Planes Connection String (MDCB) as global.remoteControlPlane's userApiKey, orgId, and connectionString respectively.

Reviewer: Should we quote it to indicate that the quoted part belongs to code piece? In the preview website, the formatting not clear for me. what do you think? If it is not important, just skip it.
```suggestion
2. Copy Key, Org ID, and Data Planes Connection String (MDCB) as `global.remoteControlPlane`'s userApiKey, orgId, and connectionString respectively.
```

{{< img src="/img/tyk-charts/tyk-cloud-deployment.png" alt="tyk-cloud-deployment" >}}

## Installing the Chart

To install the chart from the Helm repository in namespace `tyk` with the release name `tyk-data-plane`:

> ```bash```

Reviewer: Extra ``` causes incorrect formatting.

```suggestion
```bash
```
